From cb31bdc8c65dce36192d1897dc426728e62e2499 Mon Sep 17 00:00:00 2001
From: ZhaoLong Wang <wangzhaolong1@huawei.com>
Date: Thu, 16 Feb 2023 21:41:21 +0800
Subject: [PATCH 053/500] [Huawei] hikey9xx: Fixed incorrect use of kfree to
 free sreg

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188402

--------------------------------

When memory allocated by using the devm_kmalloc(). the matching
release function should use the devm_kfree() function instead of
the kfree() function.

@sreg allocated:
hi6421_spmi_regulator_probe
  hi6421_spmi_regulator_probe_ldo

@sreg freed:
hi6421_spmi_regulator_remove

Signed-off-by: ZhaoLong Wang <wangzhaolong1@huawei.com>
---
 drivers/staging/hikey9xx/hi6421v600-regulator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/hikey9xx/hi6421v600-regulator.c b/drivers/staging/hikey9xx/hi6421v600-regulator.c
index 614b03c9ddfb..63204e460f53 100644
--- a/drivers/staging/hikey9xx/hi6421v600-regulator.c
+++ b/drivers/staging/hikey9xx/hi6421v600-regulator.c
@@ -452,7 +452,7 @@ static int hi6421_spmi_regulator_remove(struct platform_device *pdev)
 	if (rdev->desc->volt_table)
 		devm_kfree(&pdev->dev, (unsigned int *)rdev->desc->volt_table);
 
-	kfree(sreg);
+	devm_kfree(&pdev->dev, sreg);
 
 	return 0;
 }
-- 
2.17.1

